src/tools/cephfs/cephfs-shell

```diff
@@ -1117,23 +1117,39 @@ sub-directories, files')
         """
         Disk Usage of a Directory
         """
+        dir_passed = args.dirs
         if args.dirs == '':
-            args.dirs = cephfs.getcwd().decode('utf-8')
-        for dir_ in args.dirs:
+            dir_passed = cephfs.getcwd().decode('utf-8')
+            if len(dir_passed) > 1:
+                dir_passed = dir_passed.split()
+
+        for dir_ in dir_passed:
+            dir_trav = sorted(set(dirwalk(dir_)))
+            if args.dirs and not dir_trav:
+                dir_trav.append(dir_)
+            else:
+                dir_trav.append('.')
+
             if args.r:
-                for i in reversed(sorted(set(dirwalk(dir_)))):
-                    i = os.path.normpath(i)
-                    try:
-                        xattr = cephfs.getxattr(to_bytes(i), 'ceph.dir.rbytes')
-                        self.poutput('{:10s} {}'.format(
-                            humansize(int(xattr.decode('utf-8'))), '.' + i))
-                    except libcephfs.Error:
+                dir_trav = reversed(dir_trav)
+
+            for i in dir_trav:
+                try:
+                    if is_file_exists(i) and not args.dirs:
```

Comment:
I have added this condition to fix printing of error message for files, when no arguments is passed to du.

Reply:
I've reported [this issue](http://tracker.ceph.com/issues/40371) and have [created a PR](https://github.com/ceph/ceph/pull/28560) for it. I personally think you should reverse this change.

Reply:
Okay, I have removed this change.